### PR TITLE
Slot machines directly use bank account balance

### PR DESCRIPTION
**Pull request by 112358sam**
[QoL] [input-wanted]

## About the PR 
Slot machines now directly use a player's bank account similar to vending machines. The withdraw to card and deposit from card functions are removed from the ATM as they were only used for slot machines. To maintain account security, slot machines now ask for a PIN before allowing play.

The slot machine UI now has the player cash in some amount from their account before play and cash out after play. Removing the card also cashes out all funds. This prevents someone with bank access from pulling someone's winnings while they are still playing.

image

## Why's this needed? 
Having two digital currencies is unintuitive and difficult for new players to learn. The withdraw to card feature is only used for slots and seems out of place with every other device using bank account funds.

## Changelog

```changelog
(u)112358sam
(+)Slot machines use bank account funds.
(+)ATMs no longer withdraw to and deposit from ID cards.
```

### CODE REVIEW

**Comment by RGBDeadSilent** [comment]
This does make it easier on new players, but it also makes it easier for antags to immediately embezzle the money directly out of your account.

HONK HONK HONK! JUSTIN FALKLIN JUST HIT IT BIG AND WON 1,000,000 CREDITS!

Let me just immediately steal that as a HoP antag, move it to somewhere I can easily pull it from, and you've now lost that money immediately.

At least the old way it was on the card, and the antag had a chance to move it to their account when they hit listening post to purchase things. (What other reasons are there to win big? Buying Station destroying pod weapons and AEX ammo thats what.)

**Comment by 112358sam** [comment]
Didn't expect this to be contentious. I'll add the input-wanted label.

File: code/obj/submachine/slots.dm [hunk 1]
```diff
@@ -93,10 +107,24 @@
...
+				usr.put_in_hand_or_eject(I)
+				ui_interact(user)
+				return
+			var/enterpin = input(user, "Please enter your PIN number.", "Enter PIN", 0) as null|num
+			if (enterpin != idcard.pin)
+				boutput(user, "<span class='alert'>Pin number incorrect.</span>")
+				usr.put_in_hand_or_eject(I)
+				ui_interact(user)
+				return
+			boutput(user, "<span class='notice'>Card authorized.</span>")
```
Comment (MarkNais) [inline code/obj/submachine/slots.dm:125]: It'd be nice to see some code-reuse in `attackby()` and `ui_act()`

**Review by MarkNais — APPROVED** [review]
As much as instant-embezzlement is a possibility, I personally don't find it to be a blocker.

File: code/obj/submachine/slots.dm (hunk 1 above)
Comment (112358sam) [reply code/obj/submachine/slots.dm]: Thanks, I've adjusted it so now ui_act calls the attackby function when inserting a card.

**Comment by 112358sam** [comment]
I've changed the slot machine to locally hold play funds until the card is ejected. I've edited the description accordingly.

**Review by MarkNais — APPROVED** [review]
Stored Funds is a neat idea. I assume you tested the changes.

**Comment by 112358sam** [comment]
> 
> 
> Stored Funds is a neat idea. I assume you tested the changes.

Yeah, I loaded it up and tested a variety of cases to make sure everything worked as intended.

**Pull request merged**